sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/Message.java

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.Context;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.azure.core.amqp.AmqpMessageConstant.ENQUEUED_TIME_UTC_ANNOTATION_NAME;
+import static com.azure.core.amqp.AmqpMessageConstant.PARTITION_KEY_ANNOTATION_NAME;
+import static com.azure.core.amqp.AmqpMessageConstant.PUBLISHER_ANNOTATION_NAME;
+import static com.azure.core.amqp.AmqpMessageConstant.SEQUENCE_NUMBER_ANNOTATION_NAME;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * The data structure encapsulating the message being sent-to and received-from Service Bus. Each Service Bus entity can
+ * be visualized as a stream of {@link Message}.
+ *
+ * <p>
+ * Here's how AMQP message sections map to {@link Message}. For reference, the specification can be found here:
+ * <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-complete-v1.0-os.pdf">AMQP 1.0 specification</a>
+ *
+ * <ol>
+ * <li>{@link #getProperties()} - AMQPMessage.ApplicationProperties section</li>
+ * <li>{@link #getBody()} - if AMQPMessage.Body has Data section</li>
+ * </ol>
+ *
+ * <p>
+ * Serializing a received {@link Message} with AMQP sections other than ApplicationProperties (with primitive Java
+ * types) and Data section is not supported.
+ * </p>
+ *
+ * @see MessageBatch
+ */
+public class Message {
+    /*
+     * These are properties owned by the service and set when a message is received.
+     */
+    static final Set<String> RESERVED_SYSTEM_PROPERTIES;
+
+    private final Map<String, Object> properties;
+    private final byte[] body;
```

Comment:
>private final byte[] body; [](start = 4, length = 26)

Accepting only bytes as body will break interoperability with other clients sending messages to Service Bus. If an application uses any standard AMQP client or another Service Bus client to send a non-byte message, that message cannot be represented by this class and bytes will not mean anything to a receiving application. We started the V1 Java SDK with bytes only message and then added support for other message types. If the V2 SDK is supporting only bytes means starting from square one again.

Reply:
I am adding String , contentType, byte[] constructor for now and will compare with what else is needed. I will add more add needed.

Reply:
It might be worth creating an issue for it so we don't lose track of this.

Reply:
Created https://github.com/Azure/azure-sdk-for-java/issues/8384